drivers/metal/rendering_device_driver_metal.mm

```diff
@@ -2384,6 +2437,11 @@ void deserialize(BufReader &p_reader) {
 			ERR_FAIL_V_MSG(ShaderID(), "Unexpected end of buffer");
 	}
 
+	// We need to regenerate the shader if the cache is moved to an incompatible device.
+	ERR_FAIL_COND_V_MSG(device_properties->features.argument_buffers_tier < MTLArgumentBuffersTier2 && binary_data.uses_argument_buffers(),
+			ShaderID(),
+			"Shader was generated with argument buffers, but device has limited support");
+
```

Comment:
This ensures that if someone copies the Godot `shader_cache` to a device that supports a lower argument buffer tier, that it won't crash, but regenerate the shaders.